pkg/cmd/server/api/v1/types.go

```diff
@@ -1327,6 +1332,11 @@ type AdmissionConfig struct {
 
 // ControllerConfig holds configuration values for controllers
 type ControllerConfig struct {
+	// LockServiceName instructs the controllers to use the endpoints of the named service in
+	// the kube-system namespace to coordinate the lock. This overrides the behavior of
+	// the controllerTTL value, and will instead use the leader election flags defined in
+	// the Kubernetes controllerArguments field.
+	LockServiceName *string `json:"lockServiceName"`
```

Comment:
maybe avoid putting "service" in the config field, and instead specify the resource to avoid needing a new field once configmaps are supported (c.f. https://github.com/kubernetes/kubernetes/issues/44857)?